cmake/O2Dependencies.cmake

```diff
@@ -162,6 +161,7 @@ o2_define_bucket(
 
     INCLUDE_DIRECTORIES
     ${FAIRROOT_INCLUDE_DIR}
+    ${FAIRROOT_INCLUDE_DIR}/fairmq # temporary fix, until bucket system works with imported targets
```

Comment:
The `dev` branch of FairRoot now exports and installs all FairMQ targets (which also get installed by `make install`, so without unit tests) in a file `${FAIRROOTPATH}/include/cmake/FairMQ.cmake`. The AliceO2 build system calls `find_package(FairRoot)` somewhere which executes the cmake script `${FAIRROOTPATH}/share/fairbase/modules/FindFairRoot.cmake` which [looks](https://github.com/FairRootGroup/FairRoot/blob/dev/cmake/modules/FindFairRoot.cmake#L45-L52) for the exported FairMQ targets.

This has the benefit that link dependencies and include directories are shipped together with the exported/imported target. The file `${FAIRROOTPATH}/include/cmake/FairMQ.cmake` contains e.g.
```cmake
(...)
# Create imported target FairRoot::FairMQ
add_library(FairRoot::FairMQ SHARED IMPORTED)

set_target_properties(FairRoot::FairMQ PROPERTIES
  INTERFACE_INCLUDE_DIRECTORIES "${_IMPORT_PREFIX}/include/fairmq"
  INTERFACE_LINK_LIBRARIES "dl;pthread;Boost::boost;Boost::log;Boost::log_setup;Boost::program_options;Boost::thread;Boost::system;Boost::filesystem;Boost::regex;Boost::date_time"
)
(...)
```
As you can see from this PR you can simplify the dependency management quite a lot and give proper control over path configurations to the upstream team. In a future patch FairRoot might also export the non-FairMQ targets in this way - FairMQ is a first test to see, if it is useful and works.

In the case where we generate the `rootcint` command in the function `O2_ROOT_GENERATE_DICTIONARY`, we have to generate the `-I` options ourselves. Currently, the bucket dependency management does not know about [target properties](https://cmake.org/cmake/help/v3.0/manual/cmake-properties.7.html#properties-on-targets) and therefore does not generate the correct `-I` option from the imported `FairRoot::FairMQ` target property `INTERFACE_INCLUDE_DIRECTORIES`. That is the reason for the "hack" in line 164 above.

The bucket dependency resolver (cmake function `GET_BUCKET_CONTENT`) can probably be extended so that it looks up the relevant target properties via [get_target_property](https://cmake.org/cmake/help/v3.0/command/get_target_property.html).